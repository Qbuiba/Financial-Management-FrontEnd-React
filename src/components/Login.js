//Login.js
import React, {useState} from 'react';
import { Link } from 'react-router-dom';
import authService from '../services/authService';
import PropTypes from 'prop-types';
import '../css/Auth.css';

function Login({onLogin}){
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
    const [message, setMessage] = useState('');
  
    const handleLogin = async (e) => {
      e.preventDefault();
      setMessage('');
      try {
        const response = await authService.login(username, password);
        setMessage('Login successful');
        console.log(response);
        onLogin(); // Trigger the authentication state change
      } catch (error) {
        setMessage('Login failed: Invalid credentials');
        console.error(error);
      }
    };
    return (
      <div className="auth-container">
        <div className="auth-box">
          <h2>Login</h2>
          <form onSubmit={handleLogin}>
            <label>Username</label>
            <input
              type="text"
              value={username}
              onChange={(e) => setUsername(e.target.value)}
            />
            <label>Password</label>
            <input
              type="password"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
            />
            <button type="submit">Login</button>
          </form>
          {message && <p className="error-message">{message}</p>}
          <p className="info-message">
            If you don't have an account, <Link to="/register">register here</Link>.
          </p>
        </div>
      </div>
    );
    }

    Login.propTypes = {
      onLogin: PropTypes.func.isRequired,
    };
    
    export default Login;