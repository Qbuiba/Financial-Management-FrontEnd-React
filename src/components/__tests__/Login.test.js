import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Login from '../Login';

describe('Login Page', () => {
  test('renders the Username label', () => {
    render(
      <MemoryRouter>
        <Login onLogin={() => {}} />
      </MemoryRouter>
    );

    const usernameLabel = screen.getByText('Username');
    expect(usernameLabel).toBeInTheDocument();
  });
});
