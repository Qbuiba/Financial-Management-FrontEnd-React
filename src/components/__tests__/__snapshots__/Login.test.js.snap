// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="auth-container"
  >
    <div
      class="auth-box"
    >
      <h2>
        Login
      </h2>
      <form>
        <label>
          Username
          <input
            type="text"
            value=""
          />
        </label>
        <label>
          Password
          <input
            type="password"
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Login
        </button>
      </form>
      <p
        class="info-message"
      >
        If you don't have an account, 
        <a
          href="/register"
        >
          register here
        </a>
        .
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Login Page renders Login page correctly 1`] = `
<DocumentFragment>
  <div
    class="auth-container"
  >
    <div
      class="auth-box"
    >
      <h2>
        Login
      </h2>
      <form>
        <label>
          Username
          <input
            type="text"
            value=""
          />
        </label>
        <label>
          Password
          <input
            type="password"
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Login
        </button>
      </form>
      <p
        class="info-message"
      >
        If you don't have an account, 
        <a
          href="/register"
        >
          register here
        </a>
        .
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test render Login page 1`] = `
<DocumentFragment>
  <div
    class="auth-container"
  >
    <div
      class="auth-box"
    >
      <h2>
        Login
      </h2>
      <form>
        <label>
          Username
          <input
            type="text"
            value=""
          />
        </label>
        <label>
          Password
          <input
            type="password"
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Login
        </button>
      </form>
      <p
        class="info-message"
      >
        If you don't have an account, 
        <a
          href="/register"
        >
          register here
        </a>
        .
      </p>
    </div>
  </div>
</DocumentFragment>
`;
